openfl/utilities/fed_timer.py

```diff
@@ -0,0 +1,61 @@
+import logging
+import os
+import sys
+import time
+from threading import Thread
+
+logger = logging.getLogger(__name__)
+
+class CustomThread(Thread):
+    def __init__(self, group=None, target=None, name=None,
+                    args=(), kwargs={}):
+        Thread.__init__(self, group, target, name, args, kwargs)
+        self._result = None
+        self._duration = None
+
+    def run(self):
+        if self._target is not None:
+            start = time.time()
+            self._result = self._target(*self._args, **self._kwargs)
+            self._duration = round(time.time() - start, 3)
+
+    def result(self):
+        return self._result, self._duration
+
+def parameterize(dec):
+    def layer(*args, **kwargs):
+        def repl(f):
+            return dec(f, *args, **kwargs)
+        return repl
+    return layer        
+
+@parameterize
+def fedtiming(func, timeout):
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        logger.info(f"Started: ({func.__name__})")
+
+        task = CustomThread(target=func, name=func.__name__, args=args, kwargs=kwargs)
```

Comment:
async methods run in the event loop in its own background context.
Is there a need to timeout and terminate the async methods? 

Although, I didn't try timeout decorator on any async method. Any specific method you have in mind that i can check?

Reply:
I am asking because one of the options for the aggregator service transformation was to make it asynchronous.
And if you look at the Director service, some of its RPCs are async. 
I am just wondering if there is a universal solution + could you specify in the PR descriptions what functions in out codebase could be targeted by this decorator?